tests/lib/bin/version-tool

```diff
@@ -0,0 +1,323 @@
+        help="run program's unit test suite and exit",
+    ):
+        # type: (Text, Text, Text, Text) -> None
+        super(_UnitTestAction, self).__init__(
+            option_strings=option_strings,
+            dest=dest,
+            default=default,
+            nargs="...",
+            help=help,
+        )
+
+    def __call__(self, parser, ns, values, option_string=None):
+        # type: (ArgumentParser, Namespace, Union[str, Sequence[Any], None], Optional[Text]) -> None
+        # We allow the caller to provide the test to invoke by giving
+        # --run-unit-tests a set of arguments.
+        argv = [sys.argv[0]]
+        if isinstance(values, list):
+            argv += values
+        unittest.main(argv=argv)
+        parser.exit()
+
+
+def consistent_relation(rel_op, delta):
+    # type: (Text, int) -> bool
+    """
+    consistent_relation returns true if the relation is consistent with delta.
+
+    The relation operator is one of ==, !=, <, <=, > or >=.
+    Delta is either 0, a positive or a negative number.
+    """
+    if rel_op == "==":
+        return delta == 0
+    elif rel_op == "!=":
+        return delta != 0
+    elif rel_op == ">":
+        return delta > 0
+    elif rel_op == ">=":
+        return delta >= 0
+    elif rel_op == "<":
+        return delta < 0
+    elif rel_op == "<=":
+        return delta <= 0
+    raise ValueError("unexpected relational operator " + rel_op)
+
+
+class ConsistentRelationTests(unittest.TestCase):
+    def test_eq(self):
+        # type: () -> None
+        self.assertFalse(consistent_relation("==", -1))
+        self.assertTrue(consistent_relation("==", 0))
+        self.assertFalse(consistent_relation("==", +1))
+
+    def test_ne(self):
+        # type: () -> None
+        self.assertTrue(consistent_relation("!=", -1))
+        self.assertFalse(consistent_relation("!=", 0))
+        self.assertTrue(consistent_relation("!=", +1))
+
+    def test_gt(self):
+        # type: () -> None
+        self.assertFalse(consistent_relation(">", -1))
+        self.assertFalse(consistent_relation(">", 0))
+        self.assertTrue(consistent_relation(">", +1))
+
+    def test_ge(self):
+        # type: () -> None
+        self.assertFalse(consistent_relation(">=", -1))
+        self.assertTrue(consistent_relation(">=", 0))
+        self.assertTrue(consistent_relation(">=", +1))
+
+    def test_lt(self):
+        # type: () -> None
+        self.assertTrue(consistent_relation("<", -1))
+        self.assertFalse(consistent_relation("<", 0))
+        self.assertFalse(consistent_relation("<", +1))
+
+    def test_le(self):
+        # type: () -> None
+        self.assertTrue(consistent_relation("<=", -1))
+        self.assertTrue(consistent_relation("<=", 0))
+        self.assertFalse(consistent_relation("<=", +1))
+
+    def test_unknown(self):
+        # type: () -> None
+        with self.assertRaises(ValueError):
+            consistent_relation("???", 0)
+
+
+def numeric_part_of(text):
```

Comment:
a comment in this function explaining it doesn't work for negative numbers would be nice

Reply:
Done